### PR TITLE
Add test for sending a command to an unsupported endpoint

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
#### Problem
We have no test coverage for sending a command to an unsupported endpoint.

#### Change overview
Add a test.

#### Testing
Ran Darwin tests and `test_suites.sh` manually, verified the test fails.  And will keep failing until https://github.com/project-chip/connectedhomeip/pull/8054 merges.

### CODE REVIEW

File: examples/chip-tool/commands/tests/Commands.h [hunk 1]
```diff
@@ -7829,6 +7832,75 @@ class TestCluster : public TestCommand
...
+        {
+            ChipLogError(chipTool, "Error: The test was expecting a success callback. Got failure callback");
+            runner->SetCommandExitStatus(CHIP_ERROR_INTERNAL);
+            return;
+        }
+
+        runner->NextTest();
+    }
+
+    static void OnTestSendClusterTestClusterCommandTest_101_SuccessResponse(void * context)
```
Comment (Damian-Nordic) [inline examples/chip-tool/commands/tests/Commands.h:7886]: Looks like this pattern repeats for other test cases as well. Could this callback functions be common for all of them? Is there a reason we can't have a common `runner->mIsFailureExpected` member?

**Review by Damian-Nordic — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

File: examples/chip-tool/commands/tests/Commands.h [hunk 2]
```diff
@@ -7829,6 +7832,75 @@ class TestCluster : public TestCommand
 
         runner->NextTest();
     }
+
+    // Test Send Test Command to unsupported endpoint
+    typedef void (*SuccessCallback_101)(void * context);
+    chip::Callback::Callback<SuccessCallback_101> * mOnSuccessCallback_101    = nullptr;
```
Comment (mspang) [inline examples/chip-tool/commands/tests/Commands.h:7838]: Why does this not say:

```
chip::Callback::Callback<SuccessCallback_101> mOnSuccessCallback_101 { &OnTestSendClusterTestClusterCommandTest_101_FailureResponse, this };
```

And if it needs to be nullable for some reason (I'm not sure why..) we could use Optional<> or std::unique_ptr.

I think this is worse than all of 3 other possibilities.

**Review by mspang — CHANGES_REQUESTED** [review]

File: examples/chip-tool/commands/tests/Commands.h [hunk 3]
```diff
@@ -7829,6 +7832,75 @@ class TestCluster : public TestCommand
 
         runner->NextTest();
     }
+
+    // Test Send Test Command to unsupported endpoint
+    typedef void (*SuccessCallback_101)(void * context);
```
Comment (mspang) [inline examples/chip-tool/commands/tests/Commands.h:7837]: nit: Prefer `using`as it is more readable than `typedef`

**Review by mspang — CHANGES_REQUESTED** [review]

File: examples/chip-tool/commands/tests/Commands.h (hunk 2 above)
Comment (bzbarsky-apple) [reply examples/chip-tool/commands/tests/Commands.h:7838]: > Why does this not say:

I'm happy to look into this, but this is auto-generated code (see the two separate commits which break out the manual changes and the generated changes) so the fix would be pretty orthogonal and would affect all of the test commands, presumably.

File: examples/chip-tool/commands/tests/Commands.h (hunk 3 above)
Comment (bzbarsky-apple) [reply examples/chip-tool/commands/tests/Commands.h:7837]: Again, this is generated code.  Happy to look into this in a followup, but don't think we should be including changes to the generator templates in this PR.

File: examples/chip-tool/commands/tests/Commands.h (hunk 1 above)
Comment (bzbarsky-apple) [reply examples/chip-tool/commands/tests/Commands.h:11121]: Possibly.  This is auto-generated code; I'd have to go look at the template to see whether we could do that and whether the resulting template would be more complicated or not as a result....

**Review by mspang — APPROVED** [review]

File: examples/chip-tool/commands/tests/Commands.h (hunk 1 above)
Comment (bzbarsky-apple) [reply examples/chip-tool/commands/tests/Commands.h:11121]: The template has various conditions in it; depending on the exact way the test is configured in the yaml you get different signatures.  So commoning up the functions is not really very simple.

Similarly, `mIsFailureExpected` is on a per-test basis, and the runner is a single runner across a suite of tests, so we can't just have one member on it to track the per-test state.

File: examples/chip-tool/commands/tests/Commands.h (hunk 2 above)
Comment (bzbarsky-apple) [reply examples/chip-tool/commands/tests/Commands.h:11073]: So there is no reason this _has_ to be nullable.  That said, right now we allocate the callback for a test right before we start that test, and deallocate it when the test finishes.  If we just made them non-nullable members we'd have them all allocated all the time, effectively.  That's probably not a problem for the sorts of devices that run chip-tool....

**Pull request merged**